### ISSUE TITLE
Respect preload setting when resizing in reader enabled

### ISSUE THREAD

**Issue by anonstash** [Difegue/LANraragi#975]
Currently it seems that images are not preloaded and/or pre-resized when resize in reader is enabled. Advancing the page results in a noticeable lag every time. It would be a significantly better user experience if the same number of "preload" images were resized ahead of time (assuming they require resizing) to reduce this reading lag. Another small thing to note is that it seems resized images are not kept in the cache after exiting the reader and returning, but it would be better to just leave them there.

**Comment by anonstash** [comment]
After reading the code, I see why resized images aren't loaded from cache, as there is currently no logic in `Model::serve_page` to check whether or not the resized image already exists before serving:
```
# This API can only serve files from the temp folder
    if ( index( $abspath, $tempfldr ) != -1 ) {

        # Apply resizing transformation if set in Settings
        if ( LANraragi::Model::Config->enable_resize ) {

            # Store resized files in a subfolder of the ID's temp folder
            my $resized_file = "$tempfldr/$id/resized/$path";
            my ( $n, $resized_folder, $e ) = fileparse( $resized_file, qr/\.[^.]*/ );
            make_path($resized_folder);

            $logger->debug("Copying file to $resized_folder for resize transformation");
            cp( $file, $resized_file ); # At this point the existing file is overwritten
```
However, I can't seem to understand why there is a lag when advancing the page as I don't see any logic which would prevent preloading pages when resize in reader is enabled. I'm currently under the impression that in `reader.js` the `src` used by `loadImage()` results in a call to `Model::serve_page`, is that not the case?

**Comment by Difegue** [comment]
Thanks for noticing! I've added some logic to try serving existing resized files if they're present.  

As for the preload issues, you're correct that the src attribute is what triggers serve_page; I know we don't precache the HEAD request we do to get size/etc, but I don't think that should cause any issues with resized versions.. I'll try to dig a bit more.

**Comment by anonstash** [comment]
Thanks for that fix! Tested it on the nightly and resized images are being sent from the cache now.
I tried to do some more digging as to why there's a lag when advancing the page and here's what I've noticed:

- The correct number of resized images show up in the server's temp folder ahead of time (based on the preload setting)
- There's a lag in the client when advancing the page only when the page has never been viewed before by _any_ client (browser). If the page has already been viewed by some client, any subsequent view by another client will not experience the lag.
- If jumping directly to a page which has already been resized but never viewed before, there is no lag. Additionally, going _backwards_ from this page does _not_ cause any lag.

Hopefully this can help narrow it down a bit. If I get time I can also help take a look at the code.
